docs/proposals/11-scheduling.md

```diff
@@ -0,0 +1,162 @@
+condensed derivation of a `Machine` containing the requested resources like `NetworkInterfaces` and the attached 
+`MachineClass` resource list. The `IRI` will be extended to be able to manage the `Reservation`s. Once the 
+`Reservation` hits a pool provider (e.g. `libvirt-provider`), the decision can be made if the `Reservation` can be 
+fulfilled or not. In case of accepting the `Reservation`, resources needs to be blocked until: 
+1. the reservation is being deleted 
+2. the corresponding `Machine` is being placed on the pool provider.
+
+The status of the `Reservation` is being propagated up to the layer where it was created. As soon as the root 
+reservation has a populated status which contains a list of possible pools, the scheduler can pick one, set the 
+`Machine.spec.machinePoolRef` and the `poollet` will broker the `Machine` to the next level.
+
+`Reservation` resource: 
+```
+apiVersion: core.ironcore.dev/v1alpha1
+kind: Reservation
+metadata:
+  namespace: default
+  name: reservation
+spec:
+  # Pools define each pool where resources should be reserved
+  pools: 
+    - poolA
+    - poolB
+    - poolC
+  # Resources defines the requested resources in a ResourceList
+  resources:
+    localStorage: 10G
+    nics: 2
+    cpu: 2
+status:
+  # Pools describe where the resource reservation was possible
+  pools: 
+    - name: poolA
+      state: Accepted
+    - name: poolB
+      state: Rejected
+```
+
+Added `IRI` methods:
+
+```protobuf
+rpc ListReservations(ListReservationsRequest) returns (ListReservationsResponse) {};
+rpc CreateReservation(CreateReservationRequest) returns (CreateReservationResponse) {};
+rpc DeleteReservation(DeleteReservationRequest) returns (DeleteReservationResponse) {};
+```
+
+
+### Detailed flow
+
+The flow to assign a `Machine` to a `Pool` consists of 3 Phases:  
+
+#### 1. Reservation Flow
+1. If `.spec.machinePoolRef` is not set, the scheduler creates a `Reservation` that includes the required resources.
+2. `poollet`s which match `.spec.pools` of `Reservation` pick it up and broker it one layer down
+3. `provider` evaluates the `Reservation` and sets its state to `Accepted` or `Rejected`, which is then 
+   propagated up the hierarchy.
+
+#### 2. Scheduling Flow
+1. On every layer, the scheduler uses a `Reservation` corresponding to a `Machine` to select a `MachinePool`. It 
+   will pick one of the `Accepted` pools of the `Reservation.status.pools` and updates the `Machine.spec.
+   machinePoolRef`.
+2. `poollet` picks up the `Machine` since `.spec.machinePoolRef` is set
+3. Once a `Machine` reaches a `provider` with a relating `Reservation`, the `Reservation` will be replaced through
+   the `Machine`. `Reservation` status will be updated to `Bound` and `poollet`s pulls the status up until to the 
+   root `Reservation`.
+
+#### 3.  Cleanup Flow
+1. The scheduler deletes root `Reservation` if `Machine` has `.spec.machinePoolRef` and if `Reservation` has a 
+   `Bound` state.
```

Comment:
should we describe the possible states for a `reservation` and also for the decision of a provider in `.status.pools[].state`?